infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientServerRouter.java

```diff
@@ -511,8 +511,6 @@ public void receive(CorfuMessage.ResponseMsg msg) {
                             .input(new LogReplicationSinkEvent(LogReplicationSinkEvent
                                     .LogReplicationSinkEventType.REMOTE_LEADER_LOSS, nodeId));
                 }
-                completeRequest(msg.getHeader().getSession(), msg.getHeader().getRequestId(),
```

Comment:
Hi Laphoi, my reason is that this future here can be for a data msg in stable scenario. We don't need to complete the future (which is analogous to SOURCE thinking that it successfully replicated a msg), when we receive a leadership loss.

Sundar, on last friday, suggested keeping this incase there are some corner cases where we can have some leaks because of not completing the request. I am yet to take a look about it in detail

Reply:
Hey Sundar, LapHoi, This leak/gap seems to have been always present. Is it okay if I address this in a follow up PR?